webui/react/src/services/decoder.ts

```diff
@@ -140,18 +160,39 @@ export const jsonToExperiments = (data: unknown): Experiment[] => {
   });
 };
 
+const ioCheckpoinToCheckpoint = (io: ioTypeCheckpoint): Checkpoint => {
+  return { ...io,
+    endTime: io.end_time || undefined,
+    id: io.id,
+    startTime: io.start_time,
+    state: io.state as CheckpointState,
+    stepId: io.step_id,
+    trialId: io.trial_id,
+    uuid: io.uuid || undefined,
+    validationMetric: io.valiation_metric || undefined,
```

Comment:
**chore:** if `io.validation_metric` is `0`, even though it might be a valid metric, it will be converted to `undefined`. Maybe this instead?
```
validationMetric: io.validation_metric != null ? io.validation_metric : undefined,
```

Reply:
yeah I missed that. progress could run into the same issue as well. fixed it